test/unit/mesh-gateway-deployment.bats

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env bats
+
+load _helpers
+
+@test "meshGateway/Deployment: disabled by default" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+
+@test "meshGateway/Deployment: enabled with meshGateway.enabled true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "meshGateway/Deployment: annotations can be set" {
```

Comment:
Might be good to make sure that setting more than one annotation will be added and formatted correctly.

Reply:
We can't set multiline strings via `--set` unfortunately. I could write a `values.yaml` file to disk I guess? Might be overkill, what do you think?